app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -325,6 +325,7 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
           var target = data.event.target
           ecommerceSchema.search_results.ecommerce.items.push({
             item_id: target.getAttribute('data-ga4-ecommerce-path'),
+            item_content_id: target.getAttribute('data-ga4-ecommerce-content-id') || undefined,
```

Comment:
Slightly confused by the structure of this code but assuming there's a simple explanation. Further down on line 355 we're adding the new id for each of the items in the search results, but what's this one here? I can't make out if it's the first one, or a set of parent attributes? From the test changes it looks like we're just checking for the ones further down, so can you clarify what this one is?

Reply:
@andysellick One of them is for the object that is pushed when a search result is clicked, and the other one is for the object that is pushed on page load/the results updating with an array of all the search results

Reply:
Thanks, that makes sense. And do the changes to the tests cover both instances?

Reply:
@andysellick I believe so - there's one for the click of a search result here: https://github.com/alphagov/govuk_publishing_components/blob/6da1e132b7f0d15a1aad4d3e3dd78436d2b1014d/spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-ecommerce-tracker.spec.js#L241 

and one for the view_item_list event here:

https://github.com/alphagov/govuk_publishing_components/blob/6da1e132b7f0d15a1aad4d3e3dd78436d2b1014d/spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-ecommerce-tracker.spec.js#L152

Thanks

Reply:
Great, thanks 👍